File: javascript/matrix/matrix.js
export class Matrix {
  #rows
  #cols

  constructor(string) {
    // splitting input str on \n
    const rows = string.split('\n')
    // splitting every row on space and turning values to numbers
    this.#rows = rows.map((r) => r.split(' ').map((val) => +val))

    this.#cols = []

    for (const row of this.#rows) {
      for (let i = 0; i < row.length; i++) {
        if (!this.#cols[i]) this.#cols.push([])
        this.#cols[i].push(row[i])
      }
    }
  }

  get rows() {
    return this.#rows
  }

  get columns() {
    return this.#cols
  }
}

File: python/matrix/matrix.py
class Matrix:
    def __init__(self, matrix_string):
        self.rows = [[*map(int, nums.split())]
                     for nums in matrix_string.splitlines()]

    def row(self, index):
        return self.rows[index - 1]

    def column(self, index):
        return [self.rows[i][index - 1] for i in range(len(self.rows))]
